File: 001-100/problem003/problem003.js
/*
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600851475143 ?
*/

var number = 600851475143,
    prime = 3;

while (number % 2 == 0) {
    number /= 2;
}

while (number != 1) {
    if (number % prime == 0) {
        while (number % prime == 0) {
            number /= prime;
        }
    }
    prime += 2
};
console.log(prime-2)

File: 001-100/problem003/problem003.py
from time import time
start = time()
def largest_prime_factor(m):
    n = m
    k = 2
    while k <= n//2:  # largest prime factor will be <= n//2
        if n % k == 0:
            while n % k == 0:
                n = n //k
        k += 1
    return n
    
print(largest_prime_factor(600851475143))
print(time() - start)
